### PR TITLE
Allow for PHP8

### PR DESCRIPTION
**Pull request by Baggerone**
I've tested it against this feature branch of developer portal
https://github.com/silinternational/developer-portal/tree/feature/PHP8

### CODE REVIEW

File: tests/ApiTest.php [hunk 1]
```diff
@@ -31,8 +88,39 @@ public function testGet()
...
+    "keylessQps": 2,
+    "keylessQpd": 172800,
+    "updatedAt": 1427892638934
+  }
+}';
+
+        $wantBody = '{"protocol":"https","tokenSkewProtectionCount":3,"apiFormat":"json",' .
+            '"endPointTimeout":2,"defaultPath":"\/api","disabled":false,"strictSSL":true,' .
+            '"sendThroughApiKey":false,"sendThroughApiSig":false,"hasCapturePaths":false,' .
+            '"allowKeylessUse":false,"keylessQps":2,"keylessQpd":172800}';
```
Comment (forevermatt) [inline tests/ApiTest.php]: I don't know whether you hand-coded these values or they were templates/examples to use for this, but if you want to make these more readable, you could code them as PHP arrays and `json_encode()` them:
```suggestion
        $wantBody = json_encode([
            'protocol' => 'https',
            'tokenSkewProtectionCount' => 3,
            'apiFormat' => 'json',
            'endPointTimeout' => 2,
            'defaultPath' => '/api',
            'disabled' => false,
            'strictSSL' => true,
            'sendThroughApiKey' => false,
            'sendThroughApiSig' => false,
            'hasCapturePaths' => false,
            'allowKeylessUse' => false,
            'keylessQps' => 2,
            'keylessQpd' => 172800,
        ]);
```

Just a suggestion.

File: tests/KeyTest.php [hunk 2]
```diff
@@ -22,7 +39,24 @@ public function testList()
     public function testGet()
     {
         $client = $this->getClient(self::objectClass);
-        $api = $client->get(['id' => 'abc123']);
+        $mockBody = '{
```
Comment (forevermatt) [inline tests/KeyTest.php:42]: I'm curious: why the switch to just using mocked values? It looks like the previous version of the test(s) made actual API calls, though I could be mis-reading it.

Comment (Baggerone) [reply tests/KeyTest.php:42]: It looked to me that previoiusly it was using the mock values in (for example)
https://github.com/silinternational/apiaxle-sdk-php/blob/master/src/mock/key.php
which would be consumed by the findMock function in 
https://github.com/silinternational/apiaxle-sdk-php/blob/master/src/mock/MockResponse.php

File: tests/ApiTest.php (hunk 1 above)
Comment (Baggerone) [reply tests/ApiTest.php]: Done

File: tests/KeyTest.php (hunk 2 above)
Comment (Baggerone) [reply tests/KeyTest.php:42]: And the mock middleware would have been keeping the actual api call from happening.
https://github.com/silinternational/apiaxle-sdk-php/blob/master/src/middleware/MockSubscriber.php

Comment (forevermatt) [reply tests/KeyTest.php:42]: Okay, thanks

**Review by forevermatt — APPROVED** [review]

**Pull request merged**